v2/stack_analyses.yaml

```diff
@@ -0,0 +1,441 @@
+                    parameter and vice versa.
+        required: true
+      responses:
+        200:
+          description: Response with the unique identifier for the stack analysis
+            request
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/StackAnalysesPostResponse'
+        400:
+          description: Bad request
+          content: {}
+        401:
+          description: Request Unauthorized
+          content: {}
+  /stack-analyses/{request-id}:
+    get:
+      tags:
+      - Scan Services
+      summary: Get security and other details about an application stack and related
+        recommendations to improve the stack
+      operationId: f8a_scanner.api_v2.stack-analyses-get
+      parameters:
+      - name: request-id
+        in: path
+        description: stack analysis request id
+        required: true
+        schema:
+          type: string
+      responses:
+        200:
+          description: Detailed response for the requested application component
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/StackAnalysesGetResponse'
+        401:
+          description: Request unauthorized
+          content: {}
+        404:
+          description: Data not found
+          content: {}
+        408:
+          description: Request has timed out on server
+          content: {}
+components:
+  schemas:
+    VulnerabilityForUser:
+      title: VulnerabilityForUser
+      allOf:
+        - $ref: '#/components/schemas/ExtendedVulnerabilityFields'
+        - oneOf:
+          - $ref: '#/components/schemas/PublicVulnerability'
+          - $ref: '#/components/schemas/PrivateVulnerability'
+      description: Vulnerability details for registered users
+    VulnerabilityForAnonymous:
+      title: VulnerabilityForAnonymous
+      allOf:
+        - $ref: '#/components/schemas/PublicVulnerability'
+      description: Vulnerability details for unregistered users
+    ExtendedVulnerabilityFields:
+      title: ExtendedVulnerabilityFields
+      type: object
+      properties:
+        malicious:
+          type: boolean
+          default: false
+        patch_exists:
+          type: boolean
+        fixable:
+          type: boolean
+        exploit:
+          type: string
+        description:
+          type: string
+        fixed_in:
+          type: string
+        references:
+          type: array
+          items:
+            type: object
+            properties:
+              title:
+                type: string
+              url:
+                type: string
+      description: Extended Vulnerability details, available only to registered users
+    PublicVulnerability:
+      title: PublicVulnerability
+      description: Publicaly known Vulnerability details with associated CVES
+      type: object
+      properties:
+        cve_ids:
```

Comment:
Please refer to this comment https://github.com/fabric8-analytics/fabric8-analytics-server/pull/606#discussion_r401400335

Reply:
For public vulnerability do have CVE id associated with it.

Reply:
No, i think Krishna was saying to change it to `vulnerability_id`.

Reply:
There are 2 fields, id=> snyk id(public and private) and cve_ids => [cve,](only for public).
If you look at the response model, cve_ids field is available only for public vulnerability.

Reply:
Here Krishna has asked we should not use `cve` word 
PTAL https://github.com/fabric8-analytics/fabric8-analytics-server/pull/609#discussion_r402081401